sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_configuration.py

```diff
@@ -0,0 +1,71 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# --------------------------------------------------------------------------
+
+from typing import TYPE_CHECKING
+
+from azure.core.configuration import Configuration
+from azure.core.pipeline import policies
+
+from ._version import VERSION
+from ._policies import JwtCredentialPolicy, ApiManagementProxy
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from typing import Any, Optional, Union
+
+    from azure.core.credentials import TokenCredential, AzureKeyCredential
+
+
+def _proxy_policy(**kwargs):
+    if kwargs.get('proxy_endpoint'):
+        return ApiManagementProxy(kwargs.get('endpoint'), kwargs.get('proxy_endpoint'))
+    return None
+
+
+class WebPubSubServiceClientConfiguration(Configuration):
+    """Configuration for WebPubSubServiceClient.
+
+    Note that all parameters used to create this instance are saved as instance
+    attributes.
+
+    :param credential: Credential needed for the client to connect to Azure.
+    :type credential: Union[~azure.core.credentials.TokenCredential, ~azure.core.credentials.AzureKeyCredential]
+    """
+
+    def __init__(
+        self,
+        credential,  # type: Union[TokenCredential, AzureKeyCredential]
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        super(WebPubSubServiceClientConfiguration, self).__init__(**kwargs)
+
+        self.credential = credential
+        self.credential_scopes = kwargs.pop('credential_scopes', ['https://webpubsub.azure.com/.default'])
+        kwargs.setdefault('sdk_moniker', 'messaging-webpubsubservice/{}'.format(VERSION))
+        self._configure(**kwargs)
+
+    def _configure(
+        self,
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        self.user_agent_policy = kwargs.get('user_agent_policy') or policies.UserAgentPolicy(**kwargs)
+        self.headers_policy = kwargs.get('headers_policy') or policies.HeadersPolicy(**kwargs)
+        self.proxy_policy = _proxy_policy(**kwargs) or policies.ProxyPolicy(**kwargs)
```

Comment:
I have not done a deep-dive here, but replacing the proxy policy seems wrong to me. What if I'm using both an actual proxy with APIM behind said proxy?

Reply:
I think we could add a new policy in `azure-core`. To catch up with webpubsub service, I realize code in [class ApiManagementProxy(CustomHookPolicy)](https://github.com/Azure/azure-sdk-for-python/blob/fe01cff929f1c7f405112600eaa40d951f4e9216/sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_configuration.py#L59) temporarily.